docs/usage/reset/demo/reset.md

```diff
@@ -0,0 +1,99 @@
+# Resetting dirty form state
+
+The following kitchen-sink like example shows all three ways to reset form state:
+
+- by clicking the native reset button
+- by clicking the link, using the yielded `reset` action
+- by calling `reset` using the register pattern described above when selecting a different user
+
+```hbs template
+<div class='flex gap-8'>
+  <div class='flex-initial'>
+    Select user:
+    <ul>
+      {{#each this.users as |user index|}}
+        <li><a
+            class='cursor-pointer'
+            {{on 'click' (fn this.selectUser index)}}
+          >{{user.firstName}}
+            {{user.lastName}}</a></li>
+      {{/each}}
+    </ul>
+  </div>
+
+  <HeadlessForm
+    @data={{this.currentUser}}
+    @onSubmit={{this.handleSubmit}}
+    class='flex-1'
+    as |form|
+  >
+    {{(this.assignReset form.reset)}}
+
+    <form.Field @name='firstName' as |field|>
+      <div class='my-2 flex flex-col'>
+        <field.Label>First name</field.Label>
+        <field.Input required class='border rounded px-2' />
+        <field.Errors />
+      </div>
+    </form.Field>
+
+    <form.Field @name='lastName' as |field|>
+      <div class='my-2 flex flex-col'>
+        <field.Label>Last name</field.Label>
+        <field.Input required class='border rounded px-2' />
+        <field.Errors />
+      </div>
+    </form.Field>
+
+    <button type='submit'>Submit</button>
+    <button type='reset'>Reset</button>
+
+    <p>Click on the reset button above or
+      <a {{on 'click' form.reset}}>here</a>
+      to reset any dirty state</p>
+  </HeadlessForm>
+</div>
+```
+
+```js component
+import Component from '@glimmer/component';
+import { tracked } from '@glimmer/tracking';
+import { action } from '@ember/object';
+
+export default class MyFormComponent extends Component {
+  users = [
+    {
+      firstName: 'Jane',
+      lastName: 'Doe',
+    },
+    {
+      firstName: 'John',
+      lastName: 'Smith',
+    },
+  ];
+
+  @tracked
+  selectedIndex = 0;
+
+  resetForm;
+
+  @action
+  assignReset(reset) {
```

Comment:
Nit: maybe there should be a comment here that this is only necessary if you want to do something with the `reset` callback?